### ISSUE TITLE
Update Key Vault to Support RBAC Permissions and Delete Protection

### ISSUE THREAD

**Issue by AErmie** [microsoft/finops-toolkit#1067]
## ⚠️ Problem
For organizations that utilize the [Enterprise Scale](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale) in association with the [Cloud Adoption Framework (CAF)](https://github.com/aztfmod/terraform-azurerm-caf), there are various Azure Policies and Initiatives that are deployed. 

One of those Initiatives is the **Enforce recommended guardrails for Azure Key Vault**. This initiative contains several policies, including:

- Azure Key Vault should use RBAC permission model
- Key vaults should have deletion protection enabled

The current deployment of the FinOps Toolkit (Hub architecture), violates those 2 policies and prevents its deployment. We have to add/create an exception in the target Subscription / Resource Group, for the deployment to complete successfully.

## 🛠️ Solution

Update the FinOps Toolkit's Key Vault implementation to support the [RBAC permissions](https://learn.microsoft.com/en-us/azure/key-vault/general/rbac-guide?tabs=azure-cli) model, and also enable [delete protection](https://learn.microsoft.com/en-us/azure/key-vault/general/soft-delete-overview). 

## ℹ️ Additional context
The client I am currently assisting, uses the CAF / Enterprise Scale Terraform modules.

## 🙋‍♀️ Ask for the community
We could use your help:
1. Please vote this issue up (👍) to prioritize it.
2. Leave comments to help us solidify the vision.

**Comment by flanakin** [comment]
Makes sense. I'm not familiar with these options, but we can probably look at this in December. If you're familiar with bicep and would like to see this sooner, we'd welcome a PR to add this. Or if you know the settings that need to be applied, you could share them here, which might speed things up. Just let us know.

Either way, thanks for the feedback!

**Comment by flanakin** [comment]
@allcontributors please add AErmie for feature

**Comment by AErmie** [comment]
@flanakin, I believe the REST API properties are: 
- `enableRbacAuthorization` 
- `enableSoftDelete`
- `enablePurgeProtection`

See: [https://learn.microsoft.com/en-us/rest/api/keyvault/keyvault/vaults/create-or-update?view=rest-keyvault-keyvault-2022-07-01&tabs=HTTP#vaultproperties](https://learn.microsoft.com/en-us/rest/api/keyvault/keyvault/vaults/create-or-update?view=rest-keyvault-keyvault-2022-07-01&tabs=HTTP#vaultproperties)

**Comment by ankurshukla03** [comment]
https://github.com/microsoft/finops-toolkit/pull/1349 - Added the PR request to enable purge protection for the keyvault as per this issue
